dev/preview/previewctl/cmd/install_context.go

```diff
@@ -19,25 +19,53 @@ var (
 
 func installContextCmd(logger *logrus.Logger) *cobra.Command {
 
+	// Used to ensure that we only install contexts
+	var lastSuccessfulPreviewEnvironment *preview.Preview = nil
+
+	install := func(timeout time.Duration) error {
+		p, err := preview.New("", logger)
+
+		if err != nil {
+			return err
+		}
+
+		if lastSuccessfulPreviewEnvironment != nil && lastSuccessfulPreviewEnvironment.Same(p) {
+			logger.Infof("The preview envrionment hasn't changed")
+			return nil
+		}
+
+		err = p.InstallContext(true, timeout)
+		if err == nil {
+			lastSuccessfulPreviewEnvironment = p
+		}
+		return err
+	}
+
```

Comment:
This part looks a bit like an anti-pattern for the pattern we established 😬 

To keep every logic testable, I believe we that code inside the `cmd` package should do nothing more than declare the CLI commands and calling functions from other packages.

If we move this part to a separate function the `preview` package, we can create tests for it 🙂